### DISCUSSION TITLE
How have a referenced figure in code?

### DISCUSSION THREAD

**Discussion by uben0** [typst/typst#2253] [Q&A] [answered]
```typst
#figure(image("foo.png")) <foo>
```
This works, but what if:
```typst
#{
  figure(image("foo.png")) <foo>
}
```
This does not. How can I associate a label to a figure in code mode?

**Comment by xkevio** [comment, marked as answer]
You can't attach a label in code mode. It seems to just be an implementation detail. What you can do is go back to content mode inside the code block:
```typ
#{
  [#figure(image("foo.png"))<foo>]
}
```
